src/app/application/components/app-collection/AppCollection.vue

```diff
@@ -54,41 +59,117 @@ type ChangeValue = {
 }
 
 const props = withDefaults(defineProps<{
-  total: number,
+  selectedRowId?: string | null
+  total?: number,
   pageNumber: number,
   pageSize: number,
   items: unknown[] | undefined,
   error: Error | undefined,
 }>(), {
+  selectedRowId: null,
   total: 0,
 })
 
 const emit = defineEmits<{
   (e: 'change', value: ChangeValue): void
+  (event: 'row:click', row: any): void
+  (event: 'row:selected', selectedRowId: string): void
 }>()
 
 const slots = useSlots()
 
+const currentSelectedRowId = ref<string | null>(null)
 const items = ref<unknown[] | undefined>(props.items)
 const cacheKey = ref<number>(0)
+
+const remainingSlots = computed(() => {
+  const { toolbar, ...remainingSlots } = slots
+
+  return remainingSlots
+})
+
 watch(() => props.items, () => {
   cacheKey.value++
   items.value = props.items
 })
-const click = (e: MouseEvent) => {
-  const $tr = (e.target as HTMLElement).closest('tr')
-  if ($tr) {
-    const $a = $tr.querySelector('a')
-    if ($a !== null) {
-      $a.click()
+
+watch(() => props.items, function (items) {
+  if (currentSelectedRowId.value !== null) {
+    return
+  }
+
+  if (Array.isArray(items) && items.length > 0) {
+    if (props.selectedRowId !== null) {
+      // Use `props.selectedRowId` if it actually exists in `props.items`
+      const hasItem = items.find((item) => item && typeof item === 'object' && 'id' in item && item.id === props.selectedRowId)
+
+      if (hasItem) {
+        currentSelectedRowId.value = props.selectedRowId
+      }
+    } else {
+      // Use the first item as a fallback.
+      const item = items[0]
+
+      if (item && typeof item === 'object' && 'id' in item && typeof item.id === 'string') {
+        currentSelectedRowId.value = item.id
+      }
     }
   }
+}, { immediate: true })
+
+watch(() => currentSelectedRowId.value, function (newSelectedRowId, oldSelectedRowId) {
+  if (newSelectedRowId && newSelectedRowId !== oldSelectedRowId) {
+    emit('row:selected', newSelectedRowId)
+  }
+}, { immediate: true })
+
+function getCellAttributes({ headerKey }: any) {
+  return {
+    class: {
+      [`${headerKey}-column`]: true,
+    },
+  }
+}
+
+function getRowAttributes(row: any): Record<string, string> {
+  if (!row) {
+    return {}
+  }
+
+  const attributes: Record<string, string> = {}
+
+  if ('id' in row && row.id === currentSelectedRowId.value) {
+    attributes.class = 'is-selected'
+  }
+
+  return attributes
+}
+
+function handleRowClickEvent(_event: any, row: any) {
+  currentSelectedRowId.value = row.id
+  emit('row:click', row)
```

Comment:
**Comment**: `AppCollection` should, at the very least currently, not contain the _behavior_ attached to row clicks. Specifically, navigating to the detail view on row click currently doesn’t make sense because almost none of our list views behave that way: instead, they select the item to show in the summary.